docs/ext/pydantic_autosummary/generate.py

```diff
@@ -0,0 +1,880 @@
+                continue
+        except ImportError:
+            pass
+
+        items.append(fullname)
+        if public_members is not None:
+            if modname in public_members:
+                public.append(fullname)
+        else:
+            if not modname.startswith("_"):
+                public.append(fullname)
+    return public, items
+
+
+def generate_autosummary_docs(
+    sources: list[str],
+    output_dir: str | os.PathLike[str] | None = None,
+    suffix: str = ".rst",
+    base_path: str | os.PathLike[str] | None = None,
+    imported_members: bool = False,
+    app: Any = None,
+    overwrite: bool = True,
+    encoding: str = "utf-8",
+) -> None:
+    showed_sources = sorted(sources)
+    if len(showed_sources) > 20:
+        showed_sources = showed_sources[:10] + ["..."] + showed_sources[-10:]
+
+    # ----------- pydantic_autosummary change
+    logger.info(
+        __("[pydantic_autosummary] generating autosummary for: %s")
+        % ", ".join(showed_sources)
+    )
+    if output_dir:
+        logger.info(__("[pydantic_autosummary] writing to %s") % output_dir)
+    # ----------/ pydantic_autosummary change
+
+    if base_path is not None:
+        sources = [os.path.join(base_path, filename) for filename in sources]
+
+    template = AutosummaryRenderer(app)
+
+    # read
+    items = find_autosummary_in_files(sources)
+
+    # keep track of new files
+    new_files = []
+
+    filename_map = app.config.autosummary_filename_map if app else {}
+
+    # write
+    for entry in sorted(set(items), key=str):
+        if entry.path is None:
+            # The corresponding autosummary:: directive did not have
+            # a :toctree: option
+            continue
+
+        path = output_dir or os.path.abspath(entry.path)
+        ensuredir(path)
+
+        try:
+            name, obj, parent, modname = import_by_name(entry.name)
+            qualname = name.replace(modname + ".", "")
+        except ImportExceptionGroup as exc:
+            try:
+                # try to import as an instance attribute
+                name, obj, parent, modname = import_ivar_by_name(entry.name)
+                qualname = name.replace(modname + ".", "")
+            except ImportError as exc2:
+                if exc2.__cause__:
+                    exceptions: list[BaseException] = [*exc.exceptions, exc2.__cause__]
+                else:
+                    exceptions = [*exc.exceptions, exc2]
+
+                errors = list({f"* {type(e).__name__}: {e}" for e in exceptions})
+                # ----------- pydantic_autosummary change
+                logger.warning(
+                    __(
+                        "[pydantic_autosummary] failed to import %s.\n"
+                        "Possible hints:\n%s"
+                    ),
+                    entry.name,
+                    "\n".join(errors),
+                )
+                # ----------/ pydantic_autosummary change
```

Comment:
Just changed the logger to use `pydantic_autosummary: ` instead of `autosummary: `